doc/sphinx/practical-tools/coqide.rst

```diff
@@ -5,130 +5,165 @@
+- Run to cursor (Ctrl-Right) to run commands up to the cursor
+- Run to end (Ctrl-End) to run commands to the end of the buffer
+- Reset Coq (Ctrl-Home) to restart the Coq process
+- Interrupt to stop processing commands
+
+Commands may:
+
+- Complete successfully.  In this case, the background of the command is marked
+  with the "processed" color (green by default).
+- Complete with a warning.  In this case, the warning appears in the messages
+  panel in yellow.  The background of the command is marked with the "processed"
+  color and the text is shown in blue and underlined.  The message text is
+  available as a tooltip on the text of the command.
+- Fail with an error.  If you're stepping through the proof line by line, the
+  error message appears in the message panel in red and the command is shown
+  in red and underlined with a pink background.  If you're in async mode,
+  described in more detail later, the message appears in the *errors panel*.
+  Double click on an entry to jump to the point of the error.  Execution
+  of commands stops unless you're in async mode.
+
+.. todo: describe the ribbon (a bit broken, though)
+
+In the previous figure :ref:`CoqIDE main screen <coqide_mainscreen>`,
 the running buffer is `Fermat.v`, all commands until
 the ``Theorem`` have been already executed, and the user tried to go
 forward executing ``Induction n``. That command failed because no such
 tactic exists (names of standard tactics are written in lowercase),
 and the failing command is underlined.
 
-Notice that the processed part of the running buffer is not editable. If
-you ever want to modify something you have to go backward using the up
-arrow tool, or even better, put the cursor where you want to go back
-and use the goto button. Unlike with `coqtop`, you should never use
-``Undo`` to go backward.
-
-There are two additional buttons for navigation within the running buffer. The
-"down" button with a line goes directly to the end; the "up" button with a line
-goes back to the beginning. The handling of errors when using the go-to-the-end
-button depends on whether Coq is running in asynchronous mode or not (see
-Chapter :ref:`asynchronousandparallelproofprocessing`). If it is not running in that mode, execution
-stops as soon as an error is found. Otherwise, execution continues, and the
-error is marked with an underline in the error foreground color, with a
-background in the error background color (pink by default). The same
-characterization of error-handling applies when running several commands using
-the "goto" button.
-
-If you ever try to execute a command that runs for a long time
-and would like to abort it before it terminates, you may
-use the interrupt button (the white cross on a red circle).
+If you're not in async mode and you modify the processed part of the buffer,
+everything after that point is undone.  Unlike `coqtop`, you should not use
+:cmd:`Undo` to go backward.
```

Comment:
Do we ever recommend the use of `Undo` and/or `coqtop` in the refman?

Reply:
No, IIRC it doesn't work correctly at all.  A couple examples use it, perhaps they should be changed.

Reply:
I recall vaguely that there was a plan to deprecate and eventually remove it completely. I don't know what happened to that. I would think that the other devs are not particularly fond of keeping it around.